common/man/C/backintime.1
.TH backintime 1 "March 2025" "version 1.5.6" "USER COMMANDS"
.SH NAME
backintime \- a simple backup tool for Linux.
.PP
This is the command line tool.
The graphical tool is backintime-qt.
.SH SYNOPSIS
.B backintime
[\-\-checksum]
[\-\-config PATH]
[\-\-debug]
[\-\-delete]
[\-\-help | \-h]
[\-\-keep\-mount]
[\-\-license]
[\-\-local\-backup |
\-\-no\-local\-backup]
[\-\-no\-crontab]
[\-\-only\-new]
[\-\-profile NAME |
\-\-profile\-id ID]
[\-\-quiet]
[\-\-share\-path PATH]
[\-\-version]

{ backup | backup\-job |
benchmark-cipher [FILE-SIZE] |
check-config |
decode [PATH] |
last\-snapshot | last\-snapshot\-path |
pw\-cache [start|stop|restart|reload|status] |
remove[\-and\-do\-not\-ask\-again] [SNAPSHOT_ID] |
restore [WHAT [WHERE [SNAPSHOT_ID]]] |
shutdown |
smart\-remove |
snapshots\-list | snapshots\-list\-path |
snapshots\-path |
unmount }

.SH DESCRIPTION
Back In Time is a simple backup tool for Linux. The backup is done by taking
snapshots of a specified set of folders.
.PP
All you have to do is configure: where to save snapshots, what folders to backup.
You can also specify a backup schedule: disabled, every 5 minutes, every
10 minutes, every hour, every day, every week, every month. To configure it use
one of the graphical interfaces available (backintime-gnome or backintime-kde4).
.PP
It acts as a 'user mode' backup tool. This means that you can backup/restore only
folders you have write access to (actually you can backup read\-only folders,
but you can't restore them).
.PP
If you want to run it as root you need to use 'sudo -i backintime'.
.PP
A new snapshot is created only if something changed since the last snapshot
(if any).
.PP
A snapshot contains all the files from the selected folders (except for exclude
patterns). In order to reduce disk space it use hard\-links (if possible)
between snapshots for unchanged files. This way a file of 10MiB, unchanged for
10 snapshots, will use only 10MiB on the disk.
.PP
When you restore a file 'A', if it already exists on the file system it will be
renamed to 'A.backup.currentdate'.
.PP
For automatic backup it use 'cron' so there is no need for a daemon, but 'cron'
must be running.
.SS Modes
.IP "\fILocal\fR" 4
.RS
Store snapshots on local HDD's (internal or USB). The drive has to be mounted
before creating a new snapshot.
.RE
.IP "\fILocal encrypted\fR" 4
.RS
Store encrypted snapshots on local HDD's (internal or USB).
Back In Time uses 'encfs' with standard configuration to encrypt all data.
Please take a look at \fIA NOTE ON SECURITY\fR.
.RE
.IP "\fISSH \fR" 4
.RS
With Mode set to SSH you can store the backup on a remote host using the
SecureShellHost protocol (ssh).
The remote path will be mount local using sshfs to provide file-access for the
graphical interface and the backup process.
Rsync and other processes called during backup process will run directly on the
remote host using ssh.
.PP
To prepare your user account for ssh-mode you have to create a password-less
login to the remote host (for further information look at
http://www.debian-administration.org/articles/152).
Type in terminal 'ssh-keygen \-t rsa' hit enter for default path and enter a
passphrase for the private key.
.PP
Finally type 'ssh-copy-id \-i ~/.ssh/id_rsa.pub <REMOTE_USER>@<HOST>' and enter
your password on remote host.
.PP
In Settingsdialog you need to set the host and remote user. If you enter a
relative path (no leading / ) it will start from remote users homedir. The
password has to be the passphrase for your private key.
.PP
.B Cipher
(the algorithm used to encrypt the data during transfer)
.br
To optimize performance you can choose the cipher used by ssh. Depending on your
environment you can have a massive speed increase compared to the default cipher.
.PP
\fIbenchmark\-cipher\fR will give you an overview over which cipher is the fastest
in your environment.
.PP
If the bottleneck of your environment is the hard-drive or the network you will
not see a big difference between the ciphers. In this case you should rather
stay on 'default'.
.PP
Please read security information about the cipher before using them in untrusted
networks (Wifi, Internet). Some of them (Arcfour, 3DES, ...) should be handled
as not secure anymore.
.PP
.B "Remote Host"
.br
If your remote host is an embedded Linux NAS or any other device with limited
functions, you could run into some problems caused by feature-less commands.
For example some devices may not have hardlink support for 'cp', 'chmod'
and 'rsync'. In this case it may help to install so-called Optware or Entware
on your device if available.
.PP
.B WARNING: THIS IS ONLY FOR EXPERIENCED USERS!
.br
If you don't know how to compile packages and how to modify a Linux system you
should NOT try to do this. There is a significant chance to break your device
and make it completely unusable with the following procedure. We will not take
any warranty for this. Make a backup of your device before proceed!
You have been warned!
.PP
You should install at least packages called 'bash', 'coreutils' and 'rsync'.
You will have to change users default shell from '/bin/sh' to '/opt/bin/bash'
in '/etc/passwd'. To add '/opt/bin:/opt/sbin:' to the start of the PATH environment
you can use 'Add prefix to SSH commands' in 'Expert Options'
with 'PATH=/opt/bin:/opt/sbin:\\$PATH'.
.PP
To check if it does work you can compare the output of '/bin/cp \-\-help'
and '/opt/bin/cp \-\-help'. If 'ssh <user>@<host> cp \-\-help' called from
your PC will print the same as '/opt/bin/cp \-\-help' called on the
remote host (via interactive ssh session) you are ready to go.
.PP
If you have questions on how to install and configure the Optware please refer
to the community of your device. You can also take a look on Back In Time FAQ on
GitHub https://github.com/bit-team/backintime/blob/-/FAQ.md
.PP
If you successfully modified your device to be able to make backups over ssh,
it would be nice if you write a 'How to' on Launchpad's Answers so we can add
this to the FAQ.
.RE
.IP "\fISSH encrypted\fR" 4
.RS
Store encrypted snapshots on remote hosts using SSH. Backintime
uses 'encfs \-\-reverse' to mount the root filesystem '/'. Rsync will sync this
encrypted view of '/' to a remote host over SSH. All encoding will be done on
the local machine. So the password will never be exposed to the remote host and
you can use the (normally) more powerful processor in you local machine for
encryption instead of weak NAS CPU's. The downside on this
is 'encfs \-\-reverse' does not support 'Filename Initialization Vector Chaining'
and 'Per-File Initialization Vectors' from the standard configuration
(take a look at 'man encfs' for further information). Please take a look at
\fIA NOTE ON SECURITY\fR.
.PP
Because of all data is transferred encrypted the log output shows encrypted
filenames, too. In the Logview-Dialog you can use 'decode' option to decrypt
the paths automatically or you can use 'backintime decode' to manually
decrypt paths. Back In Time will show all snapshots decoded so you can browse
all files as normal.
.PP
Exclude does not support wildcards ('foo*', '[fF]oo', 'fo?') because after
encoding a file these wildcards can't match any more. Only separate asterisk
that match a full file or folder will work ('foo/*', 'foo/**/bar'). All other
excludes that have wildcards will be silently ignored.
.PP
Please refer to the 'SSH' section above for information on setting up the SSH
connection.
.RE
.SS Password
If 'Save Password to Keyring' is activated Back In Time will save the Password
into GnomeKeyring (Seahorse) or KDE-KWallet. Both are secure password storages
which encrypt the password with the users login-password. So they can only be
accessed if the user is logged in.
.PP
A backup cronjob during the user isn't logged in can not collect the password
from keyring. Also if the homedir is encrypted the keyring is not accessible
from cronjobs (even if the user is logged in). For these cases the password can
be cached in RAM. If 'Cache Password for Cron' is activated Back In Time will
start a small daemon in user-space which will collect the password from keyring
and provide them for cronjobs. They will never be written to the harddrive but
a user with root permissions could access the daemon and read the password.
.SS user-callback
During backup process the application can call a user callback at different steps.
This callback is "$XDG_CONFIG_HOME/backintime/user-callback"
(by default $XDG_CONFIG_HOME is ~/.config).
.PP
The first argument is the profile id (1=Main Profile, ...).
.PP
The second argument is the profile name.
.PP
The third argument is the reason:
.RS
.TP
1
Backup process begins.
.TP
2
Backup process ends.
.TP
3
A new snapshot was taken. The extra arguments are snapshot ID and snapshot path.
.TP
4
There was an error. The second argument is the error code.
.RS
Error codes:
.TP
1
The application is not configured.
.TP
2
A "take snapshot" process is already running.
.TP
3
Can't find snapshots folder (is it on a removable drive ?).
.TP
4
A snapshot for "now" already exist.
.RE
.TP
5
On (graphical) App start.
.TP
6
On (graphical) App close.
.TP
7
Mount all necessary drives.
.TP
8
Unmount all drives.
.SH OPTIONS
.TP
\-\-checksum
Force to use checksum for checking if files have been changed. This is the same
as 'Use checksum to detect changes' in Options. But you can use this to
periodically run checksums from cronjobs. Only valid with \fIbackup\fR,
\fIbackup-job\fR and \fIrestore\fR.
.TP
\-\-config PATH
Read config from PATH. Default = ~/.config/backintime/config
.TP
--debug
Show debug messages.
.TP
--delete
Restore and delete newer files which are not in the snapshot.
WARNING: deleting files in filesystem root could break your whole system!!!
Only valid with \fIrestore\fR.
.TP
\-h, \-\-help
Display a short help
.TP
\-\-keep\-mount
Don't unmount on exit. Only valid with \fIsnapshots\-path\fR, \fIsnapshots\-list\-path\fR and
\fIlast\-snapshot\-path\fR.
.TP
\-\-license
Show license
.TP
--local-backup
Create backup files before changing local files.
Only valid with \fIrestore\fR.
.TP
--no-crontab
Do not install crontab entries.
Only valid with \fIcheck-config\fR.
.TP
--no-local-backup
Temporary disable creation of backup files before changing local files.
Only valid with \fIrestore\fR.
.TP
--only-new
Only restore files which does not exist or are newer than those in destination.
Using "rsync --update" option.
Only valid with \fIrestore\fR.
.TP
\-\-profile NAME
Select profile by name
.TP
\-\-profile\-id ID
Select profile by id
.TP
\-\-quiet
Suppress status messages on standard output.
.TP
\-\-share\-path PATH
Write runtime data (locks, messages, log and mountpoints) to PATH.
.TP
\-v, \-\-version
Show version

.SH COMMANDS
.TP
backup | \-b | \-\-backup
Take a snapshot now.
.TP
backup\-job | \-\-backup\-job
Take a snapshot (if needed) depending on schedule rules (used for cron jobs).
Back In Time will run in background for this.
.TP
benchmark-cipher | \-\-benchmark-cipher [FILE-SIZE]
Show a benchmark of all ciphers for ssh transfer.
.TP
check-config
Verify the profile in config, create snapshot path and crontab entries.
.TP
decode | \-\-decode [PATH]
Decode encrypted PATH. If no PATH is given Back In Time will read paths from
standard input.
.TP
last\-snapshot | \-\-last\-snapshot
Display last snapshot ID (if any)
.TP
last\-snapshot\-path | \-\-last\-snapshot\-path
Display the path to the last snapshot (if any)
.TP
pw\-cache | \-\-pw\-cache [start|stop|restart|reload|status]
Control the Password Cache Daemon. If no argument is given the Password Cache
will start in foreground.
.TP
remove[\-and\-do\-not\-ask\-again] | \-\-remove[\-and\-do\-not\-ask\-again] [SNAPSHOT_ID]
Remove the snapshot. If SNAPSHOT_ID is missing it will be prompted. SNAPSHOT_ID
can be an index (starting with 0 for the last snapshot) or the exact SnapshotID
(19 characters like '20130606-230501-984').
\fIremove\-and\-do\-not\-ask\-again\fR will remove the snapshot immediately.
Be careful with this!
.TP
restore | \-\-restore [WHAT [WHERE [SNAPSHOT_ID]]]
Restore file WHAT to path WHERE from snapshot SNAPSHOT_ID. If arguments are
missing they will be prompted. To restore to the original path WHERE can be an
empty string '' or just press Enter at the prompt. SNAPSHOT_ID can be an index
(starting with 0 for the last snapshot) or the exact SnapshotID
(19 characters like '20130606-230501-984')
.TP
shutdown
Shutdown the computer after the snapshot is done.
.TP
smart\-remove
Remove snapshots based on the configured Smart-Remove pattern.
.TP
snapshots\-list | \-\-snapshots\-list
Display the list of snapshot IDs (if any)
.TP
snapshots\-list\-path | \-\-snapshots\-list\-path
Display the paths to snapshots (if any)
.TP
snapshots\-path | \-\-snapshots\-path
Display path where is saves the snapshots (if configured)
.TP
unmount | \-\-unmount
Unmount the profile.

.SH A NOTE ON SECURITY AND ENCFS
Because of security issues with EncFS it is planned to remove it from
Back In Time. To continue the support of encrypted snapshots it is the goal
to replace it with an alternative if possible. See this document for details
and the current state of the transition process.
.PP
    https://github.com/bit-team/backintime/doc/ENCRYPT_TRANSITION.md
.PP
The security issues are mentioned in a security audit from 2014 and not
fixed until today. The EnCFS project is not maintained anymore and his former
maintainer recommend to switch to GoCryptFS. See the following links for further readings:
.IP
EncFS Security Audit (updated blog post): https://defuse.ca/audits/encfs.htm
.IP
EncFS Security Audit (original mailing list entry): https://sourceforge.net/p/encfs/mailman/message/31849549
.IP
Back In Time Issue #1734 about transition of the encryption feature: https://github.com/bit-team/backintime/issues/1734
.SH SEE ALSO
.BR backintime-qt (1),
.BR backintime-config (1),
.BR backintime-askpass (1)
.PP
\fBBack In Time\fP project website: https://github.com/bit-team/backintime
.PP
\fBBack In Time\fP mailing list: https://mail.python.org/mailman3/lists/bit-dev.python.org
.SH AUTHOR
\fBBack In Time\fP Team


CHANGES
Back In Time

Version 1.6.0-dev (development of upcoming release)
* Changed: "Repeatedly (anacron)" scheduling behave consistent. Reversed minor bug introduced with 060324e (#1791) in 1.5.3 (#2250)
* Changed: Stricter permissions (600) for fileinfo.bz2 and takesnapshot.log.bz2 (#2235)
* Changed: **Breaking** A "snapshot" now is a "backup" (#1929)
* Changed: **Breaking** Deprecated and removed make targets: "test", "test-v", "unittest", "unittest-v"
* Changed: Unlocking ssh-agent use "force" instead of "prefer" for SSH_ASKPASS_REQUIRE (#2170) (@daviewales)
* Changed: Stop passing window ID when inhibiting suspend via D-Bus power manager (#2084)
* Changed: Reorder DBUS service provider list for suspend mode inhibition (#2084)
* Changed: Man pages generated from AsciiDoc, except backintime-config (#2085)
* Changed: Deprecate command "benchmark-cipher" (#2120)
* Changed: Deprecate command "snapshots-path" (#2130)
* Changed: Deprecate commands "snapshots-list", "snapshots-list-path", "last-snapshots" and "last-snapshot-path" (#2130)
* Changed: Deprecate command "smart-remove" (#2124)
* Changed: Deprecate command "backup-job" (#2124)
* Changed: Deprecate command "remove-and-do-not-ask-again" (#2124)
* Changed: Deprecate command "decode" (#2124)
* Changed: Deprecate flag-like command aliases (e.g. "--backup" for "backup") (#2124)
* Changed: Deprecate argument flag "--profile-id" (#2125)
* Changed: Deprecate argument flag "--share-path" (#2124)
* Changed: Deprecate use of SSH Cipher (#2143)
* Changed: Prefer ed25519 SSH keys (if present) for new created profiles (#2094)
* Changed: Generating new SSH key will use default key type provided by ssh-keygen (#2194)
* Changed: Open man page and changelog in a text dialog
* Added: CLI command "status" summarize last run and backup status (#1018, Samuel Moore @s4moore)
* Added: Root mode indicator in status bar (#1964)
* Added: Option to not using an explicit SSH key file. In consequence this supports extern key agents and SSH clients own configuration (#1146)
* Added: SSH-Key selector widget in Manage profiles dialog (#1146, #2094, #2095)
* Added: "ETA" value in status bar (#2101)
* Added: Shutdown confirmation dialog (#2102) (Huaide Jiang @LatiosInAltoMare)
* Added: Check and warn if include list entries do not exists in backup source (#1586) (@rafaelhdr)
* Added: Asciidoctor as build dependence (#2085)
* Added: Command "show" to replace "snapshots-list", "snapshots-list-path", "last-snapshots" and "last-snapshot-path" (#2130)
* Added: Command "prune" to replace "smart-remove" (#2124)
* Added: Flag "--background" for command "backup" to replace command "backup-job" (#2124)
* Added: Flag "--skip-confirmation" for command "remove" to replace command "remove-and-do-not-ask-again" (#2124)
* Added: Flag "--profile" accept ID's beside names only (#2125)
* Added: Flag "-p" as alias for "--profile" (#2125)
* Added: "Less storage space" threshold to warn the user (#2110) (@fest6)
* Added: Remember size and position of Manage profiles dialog
* Added: User-callback editor offer a default script in case no script exists (#1331)
* Removed: **Breaking** Drop Python support for version 3.9 & 3.10 (#2129)
* Removed: Stop using QWindow.winId() (#2084)
* Removed: UI translation in Bosnian, Thai, and Occidental/Interlingue (#1914)
* Removed: LICENSE file in favor of LICENSES directory and LICENSES.md file
* Removed: SSH Cipher configuration in Manage profiles dialog (#2143)
* Fixed: Use LC_ALL instead of LC_TIME to set the locale
* Fixed: Optimize subparsers and usage output (#2132)
* Fixed: Re-design about dialog (#1936)
* Fixed: Stop waking up monitor when inhibit suspend on backup starts (#714, #1090)
* Fixed: Avoid shutdown confirmation dialog on Budgie and Cinnamon desktop environments (#788)
* Fixed: Crash in "Manage profiles" dialog when using "qt6ct" (#2128)
* Fixed: Allow in BITs root-mode opening URLs in extern browse

Version 1.5.6 (2025-10-05)
* Fixed: Always use 0 as window ID value when inhibiting suspend via D-Bus power manager (#2084, #2268, #2192)
* Fixed: Crash in "Manage profiles" dialog when using "qt6ct" (#2128)
* Fixed: Open online changelog if local CHANGES file is missing (#2266)
* Fixed: Disable opening browser (e.g. project website) in root-mode

Version 1.5.5 (2025-06-05)
* Fixed: Unlocking SSH keys with passphrases on new created profiles (#2164) (@davidfjoh)

Version 1.5.4 (2025-03-24)
* Breaking Change: Auto-remove rules "Free inodes" and "Free space" disabled by default in new created profiles (#1976)
* Changed: Completed license information to conform to REUSE.software and SPDX standards.
* Changed: More clear and intense warning about EncFS deprecation and removal (#1904)
* Changed: Updated desktop entry files
* Changed: Move several values from config file into new introduce state file ($XDG_STATE_HOME/backintime.json)
* Fix!: Smart-remove rule "Keep one snapshots per week or the last week" use calendar weeks
* Fix: Exclude patterns are now case-sensitive when added (#2040)
* Fix: The width of the fourth column in files view is now saved
* Fix: Snapshot compare copy symlink as symlink (#1902) (Peter Sevens @sevens)
* Fix: Crash when comparing a snapshot with a symlink pointing to a nonexistent target (Peter Sevens @sevens)
* Fix: Crash (KeyError) opening language setup dialog with unknown locale/language
* Doc: Remove & Retention (formally known as Auto-/Smart-Remove) with improved GUI and user manual section (#2000)
* Feature: Open user manual (local if available otherwise online) via Help menu
* Feature: Toolbar context menu to display the buttons in different combinations with icons and text (#1105, #2002) (Samuel Moore @s4moore)
* Feature: Add offset minutes to hourly schedules (David Gibbs @fallingrock)

Version 1.5.3 (2024-11-13)
* Doc: User manual (build with MkDocs) (#1838) (Kosta Vukicevic @stcksmsh)
* Doc: User-callback topic in user manual (#1659)
* Feature: Support language Interlingua (Occidental)

common/backintime.py
# SPDX-FileCopyrightText: © 2008-2022 Oprea Dan
# SPDX-FileCopyrightText: © 2008-2022 Bart de Koning
# SPDX-FileCopyrightText: © 2008-2022 Richard Bailey
# SPDX-FileCopyrightText: © 2008-2022 Germar Reitze
#
# SPDX-License-Identifier: GPL-2.0-or-later
#
# This file is part of the program "Back In Time" which is released under GNU
# General Public License v2 (GPLv2). See LICENSES directory or go to
# <https://spdx.org/licenses/GPL-2.0-or-later.html>.
import os
import subprocess
import pathlib
from datetime import datetime
import tools
# Workaround for situations where startApp() is not invoked.
# E.g. when using --diagnostics and other argparse.Action
tools.initiate_translation(None)
import bitbase
import config
import logger
import cli
import cliarguments
from diagnostics import collect_minimal_diagnostics


def takeSnapshotAsync(cfg, checksum=False):
    """
    Fork a new backintime process with 'backup' command which will
    take a new snapshot in background.

    Args:
        cfg (config.Config): config that should be used
    """
    cmd = []

    if cfg.ioniceOnUser():
        cmd.extend(('ionice', '-c2', '-n7'))

    cmd.append('backintime')

    if '1' != cfg.currentProfile():
        cmd.extend(('--profile-id', str(cfg.currentProfile())))

    if cfg._LOCAL_CONFIG_PATH is not cfg._DEFAULT_CONFIG_PATH:
        cmd.extend(('--config', cfg._LOCAL_CONFIG_PATH))

    # if cfg._LOCAL_DATA_FOLDER is not cfg._DEFAULT_LOCAL_DATA_FOLDER:
    #     cmd.extend(('--share-path', cfg.DATA_FOLDER_ROOT))

    if logger.DEBUG:
        cmd.append('--debug')

    if checksum:
        cmd.append('--checksum')

    cmd.append('backup')

    # child process need to start its own ssh-agent because otherwise
    # it would be lost without ssh-agent if parent will close
    env = os.environ.copy()

    for i in ('SSH_AUTH_SOCK', 'SSH_AGENT_PID'):
        try:
            del env[i]

        except:
            pass

    subprocess.Popen(cmd, env=env)


def encfs_deprecation_warning():
    """Warn about encfs deprecation in syslog.

    See Issue #1734 for details. This function is a workraound and will be
    removed if #1734 is closed.
    """

    # Don't warn if EncFS isn't installed
    if not tools.checkCommand('encfs'):
        return

    # Timestamp file
    xdg_state = os.environ.get('XDG_STATE_HOME', None)
    if xdg_state:
        xdg_state = pathlib.Path(xdg_state)
    else:
        xdg_state = pathlib.Path.home() / '.local' / 'state'
    fp = xdg_state / 'backintime.encfs-warning.timestamp'

    # ensure existence
    if not fp.exists():
        fp.parent.mkdir(parents=True, exist_ok=True)
        fp.touch()

    # Calculate age of that file
    delta = datetime.now() - datetime.fromtimestamp(fp.stat().st_mtime)

    # Don't warn if to young
    if delta.days < 30:
        return

    logger.warning('EncFS encrypted profiles are deprecated in Back In Time. '
                   'Removal is schedule for minor release 1.7 in year 2026. '
                   'For details and alternatives '
                   f'read: {bitbase.URL_ENCRYPT_TRANSITION}')

    # refresh timestamp
    fp.touch()


def startApp(app_name='backintime'):
    """
    Start the requested command or return config if there was no command
    in arguments.

    Args:
        app_name (str): string representing the current application

    Returns:
        config.Config:  current config if no command was given in arguments
    """
    parser_agent = cliarguments.ParserAgent(
        app_name=bitbase.APP_NAME, bin_name=app_name)

    logger.openlog()

    args = cliarguments.parse_arguments(args=None, agent=parser_agent)

    # Name, Version, As Root, OS
    msg = ''
    for key, val in collect_minimal_diagnostics().items():
        msg = f'{msg}; {key}: {val}'
    logger.debug(msg[2:])

    # Add source path to $PATH environ if running from source
    if tools.runningFromSource():
        tools.addSourceToPathEnviron()

    # Warn about sudo
    if (os.getenv('SUDO_USER')  # exists only if sudo was used
            and os.getenv('BIT_SUDO_WARNING_PRINTED', 'false') == 'false'):

        os.putenv('BIT_SUDO_WARNING_PRINTED', 'true')
        logger.warning(
            "It looks like you're using 'sudo' to start "
            f"{config.Config.APP_NAME}. This will cause some trouble. "
            f"Please use either 'sudo -i {app_name}' or 'pkexec {app_name}'.")

    encfs_deprecation_warning()

    # Call commands
    if 'func' in dir(args):
        args.func(args)
        return None

    # No arguments/commands
    cli.set_quiet(args)
    cli.print_header()

    return cli.get_config_and_select_profile(
        config_path=args.config,
        data_path=args.share_path,
        profile=args.profile,
        # Dev note (buhtz, 2025): There is not a default value in all cases,
        # because "--checksum" is exclusive to rsync-related commands.
        checksum=getattr(args, 'checksum', None),
        check=False)


if __name__ == '__main__':
    startApp()
